app/scripts/modules/core/src/api/mock/expectedRequest.ts

```diff
@@ -0,0 +1,145 @@
+import { ReceivedRequest } from 'core/api/mock/receivedRequest';
+import { deferred, isSuccessStatus, UrlArg, Verb } from './mockHttpUtils';
+import { SETTINGS } from 'core/config';
+import { isEqual, isMatch } from 'lodash';
+
+function parseParams(queryString: string): Record<string, string | string[]> {
+  const paramTuples = queryString.split('&').map((param) => param.split('=')) as Array<[string, string]>;
+
+  return paramTuples.reduce((paramsObj, [key, value]) => {
+    const currentValue = paramsObj[key];
+    if (typeof currentValue === 'string') {
+      paramsObj[key] = [currentValue, value];
+    } else if (Array.isArray(currentValue)) {
+      paramsObj[key] = paramsObj[key].concat(value);
+    } else {
+      paramsObj[key] = value;
+    }
+    return paramsObj;
+  }, {} as Record<string, string | string[]>);
+}
+
+/**
+ * This internal class represents an HTTP request that is expected to be made by the code under test.
+ * Instances are created by MockHttpClient.expect -- this class should not be used directly.
```

Comment:
This is my own naiveté speaking here, but the class is exported, so is it actually an internal class at that point?

Reply:
good catch

Reply:
Javascript/Typescript doesn't have the concept of package protected, but this is what was meant by "internal"